src/http_api_handler.erl

```diff
@@ -89,21 +89,30 @@ handle_request(<<"POST">>, _, json, Req, State) ->
             {true, Req2, State}
     end;
 
+handle_request(<<"DELETE">>, <<"/api/v1/contexts">>, json, Req, State) ->
+    ok = ergw_api:delete_contexts(all),
+    Response = jsx:encode(#{contexts => 0}),
+    Req2 = cowboy_req:set_resp_body(Response, Req),
+    {true, Req2, State};
+
 handle_request(<<"DELETE">>, <<"/api/v1/contexts/", _/binary>>, json, Req, State) ->
     Value = cowboy_req:binding(count, Req),
     case catch binary_to_integer(Value) of
 	Count when is_integer(Count), Count > 0 ->
 	    ok = ergw_api:delete_contexts(Count),
-	    Contexts = ergw_api:contexts(all),
-
-	    Response = jsx:encode(#{contexts => erlang:length(Contexts)}),
+        Contexts = case length(ergw_api:contexts(all)) - Count of
+            Res when Res > 0 -> Res;
+            _ -> 0
+        end,
+	    Response = jsx:encode(#{contexts => Contexts}),
```

Comment:
Why? That makes zero sense. DELETE is asynchronous operation and not matter of pretens will change that. Return the correct values and document what they mean in the API description.